### DISCUSSION TITLE
Render a large `ui.table()` and avoid app crashes

### DISCUSSION THREAD

**Discussion by azim-b** [h2oai/wave#1126] [Q&A]
While rendering a _large_ `ui.table() `, the app crashes and the server returns a _413 - Request Entity Too Large error_. Questions:

a. depending on the # of cols in the table, the # of rows I can “safely” display changes. How can I determine this in real-time and/or avoid the app from crashing?

b. one option is to display “pages” or tabs of limited # of rows. Is there an easier alternative?

**Comment by bbarclay** [comment]
I'm curious if there is a simple solution somewhere too. Server side pagination would be awesome.

**Comment by mturoci** [comment]
> depending on the # of cols in the table, the # of rows I can “safely” display changes. How can I determine this in real-time and/or avoid the app from crashing?

You can manually compute the request size yourself, but I wouldn't go down that path.

> Is there an easier alternative?

We will have to introduce a server-side search/filter/sort, tracked in #920. Since the table is already virtualized, it seems like we just need to think of way to chunk large requests and stream them down to the client, tracked in https://github.com/h2oai/wave/issues/1146

**Comment by lo5** [comment]
Related: https://github.com/h2oai/wave/issues/730#issuecomment-988159271

**Comment by azim-b** [comment]
thanks @mturoci @lo5 

As a work around for time being, I'm displaying "pages" (with `ui.tab()`s) with 100 rows/page. Is that good practice?
